osu.Game/Overlays/OverlayColourProvider.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osuTK;
+using osuTK.Graphics;
+
+namespace osu.Game.Overlays
+{
+    public class OverlayColourProvider
+    {
+        private readonly OverlayColourScheme colourScheme;
+
+        public OverlayColourProvider(OverlayColourScheme colourScheme)
+        {
+            this.colourScheme = colourScheme;
+        }
+
+        private Color4 convert(float saturation, float lightness) => Color4.FromHsl(new Vector4(getBaseHue(colourScheme), saturation, lightness, 1));
```

Comment:
convert isn't a great method name (it's not really converting). call it `getColourFromHSL` or even just `getColour`.